crates/turborepo-lib/src/task_hash.rs

```diff
@@ -0,0 +1,318 @@
+use std::collections::{HashMap, HashSet};
+
+use thiserror::Error;
+use tokio::sync::Mutex;
```

Comment:
We should only use the tokio `Mutex` if it needs to be held across an await [and we probably should avoid writing code that does that](https://tokio.rs/tokio/tutorial/shared-state#holding-a-mutexguard-across-an-await).
```suggestion
use std::sync::Mutex;
```